zmq-lwt/src/deferred.ml

```diff
@@ -0,0 +1,24 @@
+open Lwt
+
+type 'a t = 'a Lwt.t
+let (>>=) = Lwt.(>>=)
+let return a = Lwt.return a
+let try_with f = Lwt_result.catch (f ())
+
+module Fd = struct
+  type nonrec 'a t' = 'a t
+  type t = Lwt_unix.file_descr
+
+  let create fd = Lwt_unix.of_unix_file_descr fd
+  let syscall_exn fd f = Lwt_unix.wrap_syscall Lwt_unix.Read fd f
```

Comment:
In the original implementation, we always give the argument `Lwt_unix.Read` here. I would think it should depend on the operation given, i.e. if its a read or a write.
https://github.com/hcarty/lwt-zmq/blob/master/src/lwt_zmq.ml#L22

Reply:
Based on what I've read, the `Lwt_unix.Read` argument is correct here.  I'll see if I can dig up the mailing list post(s) I originally used to figure that out.